src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -1176,3 +1185,54 @@ def test_following_preceding():
             [c_ref, d_ref, routine2, assign2, e_ref, zero])
     assert (multiply1.preceding(routine=False) ==
             [container, routine1, assign1, a_ref, multiply2, b_ref])
+
+
+def test_node_same_root():
```

Comment:
Thanks for adding the missing test. CodeCov still says that a line is not covered when one of the parents is none. Could you add at the end a `two._parent = None` and call sameRoot again. This should cover it.

Reply:
Done, fingers crossed.